Assets/MRTK/SDK/Experimental/RiggedHandVisualizer/RiggedHandVisualizer.cs

```diff
@@ -0,0 +1,254 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit;
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Experimental.RiggedHandVisualizer
+{
+    /// <summary>
+    /// Hand visualizer that controls a hierarchy of transforms to be used by a SkinnedMeshRenderer
+    /// Implementation is derived from LeapMotion RiggedHand and RiggedFinger and has visual parity
+    /// </summary>
+    public class RiggedHandVisualizer : MonoBehaviour, IMixedRealityControllerVisualizer, IMixedRealitySourceStateHandler, IMixedRealityHandJointHandler
+    {
+        public virtual Handedness Handedness { get; set; }
+
+        public GameObject GameObjectProxy => gameObject;
+
+        public IMixedRealityController Controller { get; set; }
+
+        public Transform Wrist;
+        public Transform Palm;
+        
+        // User configurable metacarpal transforms for each finger
+        public Transform ThumbMetacarpal;
+        public Transform IndexMetacarpal;
+        public Transform MiddleMetacarpal;
+        public Transform RingMetacarpal;
+        public Transform PinkyMetacarpal;
+
+        [Tooltip("Hands are typically rigged in 3D packages with the palm transform near the wrist. Uncheck this if your model's palm transform is at the center of the palm similar to Leap API hands.")]
+        public bool modelPalmAtLeapWrist = true;
+
+        [Tooltip("Allows the mesh to be stretched to align with finger joint positions. Only set to true when mesh is not visible.")]
+        public bool deformPosition = true;
+
```

Comment:
Public properties should be capitalized, also applies to the properties below, [naming guidelines mrtk](https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Contributing/CodingGuidelines.html#naming-conventions)